### PR TITLE
Unnecessary variables

### PR DESCRIPTION
**Pull request by tivervac**
Removed unnecessary assignments in toString() and hashCode(). This is especially useful to avoid false positives in static analysis.

### CODE REVIEW

File: org.eclipse.xtend.lib/xtend-gen/org/eclipse/xtend/lib/DataProcessor.java [hunk 1]
```diff
@@ -101,13 +102,13 @@ public void doTransform(final MutableClassDeclaration it, @Extension final Trans
     final ToStringProcessor.Util toStringUtil = new ToStringProcessor.Util(context);
     @Extension
     final FinalFieldsConstructorProcessor.Util requiredArgsUtil = new FinalFieldsConstructorProcessor.Util(context);
-    final Procedure1<MutableFieldDeclaration> _function = new Procedure1<MutableFieldDeclaration>() {
+    final Consumer<MutableFieldDeclaration> _function = new Consumer<MutableFieldDeclaration>() {
```
Comment (szarnekow) [inline org.eclipse.xtend.lib/xtend-gen/org/eclipse/xtend/lib/DataProcessor.java]: I'm not sure about Java8 Code here.

Comment (cdietrich) [reply org.eclipse.xtend.lib/xtend-gen/org/eclipse/xtend/lib/DataProcessor.java]: well that is that "we do still pseudo java 6 crap" i dont know who is using it :(

Comment (cdietrich) [reply org.eclipse.xtend.lib/xtend-gen/org/eclipse/xtend/lib/DataProcessor.java]: i dont even know how to run a xtend compiler that needs java 8 agains a java 6 toolchain

**Comment by cdietrich** [comment]
@tivervac  could you have a look?

**Comment by tivervac** [comment]
@cdietrich What exactly should I have a look at? I honestly assumed that we were using Java 8 already. Isn't Eclipse Luna the minimum required Eclipse version?

**Comment by cdietrich** [comment]
Yes and no for some reason I don’t understand xbase lib is intended to be Java 6

**Comment by cdietrich** [comment]
See https://github.com/eclipse/xtext-lib/issues/51

**Comment by tivervac** [comment]
So I guess I have to make sure the xtend-gen is Java6 compliant?

Why can't we try out Java 8 like we tried incrementing the guava version in #30 ?
Quoting Sven:
> Ok, we had a brief offline discussion and decided to do it. If it reveals any issues we night roll it back in a bugfix release. @ cdietrich agreed to do the changes

**Comment by cdietrich** [comment]
i have created https://github.com/eclipse/xtext-lib/pull/77. i dont think anyone uses java 6 with a current version of xtend

**Comment by cdietrich** [comment]
@tivervac we are on java8 guava21 now. can you please rebase

**Comment by cdietrich** [comment]
@eclipsewebmaster @waynebeaton can you please tell us why the ip validation fails?

**Comment by szarnekow** [comment]
LGTM

**Comment by eclipsewebmaster** [comment]
Just to update here as well as the Eclipse.org bug:

I'm pretty sure the issue here is that while the signed-off-by footer is in fact correct, the email contained in the 'committer' section of the JSON payload is @gmail rather than @sigasi(as determined by signed-off-by), and so the verification fails since these are expected to match.

Not sure if this is a result of using the GitHub UI, or because the contributors GitHub account is using a different email address then their Eclipse.org account.

Based on https://www.eclipse.org/projects/handbook/#resources-commit? this appears to be the expected behaviour.

**Comment by cdietrich** [comment]
We redid the commit and thus solved the problem. Thx for the insights

**Pull request closed without merging**